### PR TITLE
chore: add datadog tracing to tasks

### PR DESCRIPTION
**Pull request by miketheman**
Previously attempted in #12981 and reverted in #13003

ddtrace shipped a fix recently after in 1.9.0
Refs: https://github.com/DataDog/dd-trace-py/pull/4969

This time we also add a distinct name for the worker class (thanks, Karl) as Cabotage doesn't provide a facility yet to set environment variables per-Procfile process.

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 release: bin/release
 web: bin/start-web ddtrace-run python -m gunicorn.app.wsgiapp -c gunicorn-prod.conf.py warehouse.wsgi:application
 web-uploads: bin/start-web ddtrace-run python -m gunicorn.app.wsgiapp -c gunicorn-uploads.conf.py warehouse.wsgi:application
-worker: bin/start-worker celery -A warehouse worker -Q default -l info --max-tasks-per-child 32
-worker-beat: bin/start-worker celery -A warehouse beat -S redbeat.RedBeatScheduler -l info
+worker: bin/start-worker ddtrace-run celery -A warehouse worker -Q default -l info --max-tasks-per-child 32
```
Comment (ewdurbin) [inline Procfile:4]: Cabotage does allow setting envs via the Procfile, as defined in this spec: https://smartmob-rfc.readthedocs.io/en/latest/1-procfile.html#formal-specification

So you could update:

```suggestion
worker: DD_SERVICE=warehouse-worker bin/start-worker ddtrace-run celery -A warehouse worker -Q default -l info --max-tasks-per-child 32
```

File: bin/start-worker [hunk 2]
```diff
@@ -6,5 +6,8 @@ set -eo pipefail
 # Setup our Redis settings
 source bin/redis-tls
 
+# Override the project-wide
+export DD_SERVICE=warehouse-worker
+
```
Comment (miketheman) [inline bin/start-worker:11]:
```suggestion
```

File: Procfile (hunk 1 above)
Comment (miketheman) [reply Procfile]: Thanks, TIL! Updated.

File: Procfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 release: bin/release
 web: bin/start-web ddtrace-run python -m gunicorn.app.wsgiapp -c gunicorn-prod.conf.py warehouse.wsgi:application
 web-uploads: bin/start-web ddtrace-run python -m gunicorn.app.wsgiapp -c gunicorn-uploads.conf.py warehouse.wsgi:application
-worker: bin/start-worker celery -A warehouse worker -Q default -l info --max-tasks-per-child 32
-worker-beat: bin/start-worker celery -A warehouse beat -S redbeat.RedBeatScheduler -l info
+worker: DD_SERVICE=warehouse-worker bin/start-worker ddtrace-run celery -A warehouse worker -Q default -l info --max-tasks-per-child 32
```
Comment (ewdurbin) [inline Procfile:4]:
```suggestion
worker: env DD_SERVICE=warehouse-worker bin/start-worker ddtrace-run celery -A warehouse worker -Q default -l info --max-tasks-per-child 32

My bad! That's the syntax our parser consumes.
```

**Review by ewdurbin — APPROVED** [review]

**Pull request merged**